framework/src/main/java/org/tron/core/net/service/effective/ResilienceService.java

```diff
@@ -74,75 +74,80 @@ private void disconnectRandom() {
     if (peerSize >= CommonParameter.getInstance().getMaxConnections()) {
       long now = System.currentTimeMillis();
       List<PeerConnection> peers = tronNetDelegate.getActivePeer().stream()
-          .filter(peer -> !peer.isDisconnect())
           .filter(peer -> now - peer.getLastActiveTime() >= inactiveThreshold)
           .filter(peer -> !peer.getChannel().isTrustPeer())
+          .filter(peer -> !peer.isNeedSyncFromUs() && !peer.isNeedSyncFromPeer())
           .collect(Collectors.toList());
       if (!peers.isEmpty()) {
         int index = new Random().nextInt(peers.size());
-        disconnectFromPeer(peers.get(index), ReasonCode.RANDOM_ELIMINATION);
+        disconnectFromPeer(peers.get(index), ReasonCode.RANDOM_ELIMINATION,
+            DisconnectCause.RANDOM_ELIMINATION);
       }
     }
   }
 
   private void disconnectLan() {
-    if (isLanNode()) {
-      // disconnect from the node that has keep inactive for more than inactiveThreshold
-      // and its lastActiveTime is smallest
-      int peerSize = tronNetDelegate.getActivePeer().size();
-      if (peerSize >= CommonParameter.getInstance().getMinConnections()) {
-        long now = System.currentTimeMillis();
-        List<PeerConnection> peers = tronNetDelegate.getActivePeer().stream()
-            .filter(peer -> !peer.isDisconnect())
-            .filter(peer -> now - peer.getLastActiveTime() >= inactiveThreshold)
-            .filter(peer -> !peer.getChannel().isTrustPeer())
-            .collect(Collectors.toList());
-        Optional<PeerConnection> one = getEarliestPeer(peers);
-        one.ifPresent(peer -> disconnectFromPeer(peer, ReasonCode.BAD_PROTOCOL));
-      }
+    if (!isLanNode()) {
+      return;
+    }
+    // disconnect from the node that has keep inactive for more than inactiveThreshold
+    // and its lastActiveTime is smallest
+    int peerSize = tronNetDelegate.getActivePeer().size();
+    if (peerSize >= CommonParameter.getInstance().getMinConnections()) {
```

Comment:
Is this conditional statement redundant?

Reply:
It's not redundant, condition of disconnection is different:  peerSize > CommonParameter.getInstance().getMinActiveConnections().

Reply:
Why is it '>', shouldn't lan nodes be '=' in most scenarios?

Reply:
It's ">=", not ">".

Reply:
I'm talking about why the comparison in the isLanNode method is '>'

Reply:
You are right, I will revise it.

Reply:
So let's go back to the initial question: how do you think?

Reply:
Even it's a lan node, it will disconnect from some nodes only if peerSize >= CommonParameter.getInstance().getMinConnections(). Later condition is more strict. It's not redundant.